### PR TITLE
Transform UniFFI SecurityShieldBuilder to value type semantics

### PR DESCRIPTION
**Pull request by danvleju-rdx**
To improve the handling of TCA state in the iOS host, we need to transform the UniFFI `SecurityShieldBuilder` to use value type semantics, similar to how the [TransactionBuilder is implemented in radix-engine-toolkit](https://github.com/radixdlt/radix-engine-toolkit/blob/2ab5b35a6cb96c9faeb98ed8df67b870336e6873/crates/radix-engine-toolkit-uniffi/src/builder/manifest_builder/builder_v1.rs#L24).

This change means that every mutating function will return a new instance of `SecurityShieldBuilder`.

### CODE REVIEW

File: crates/sargon-uniffi/src/profile/mfa/security_structures/security_shield_builder.rs [hunk 1]
```diff
@@ -19,10 +19,33 @@ use crate::prelude::*;
 /// which contains a MatrixOfFactorSourceIds - with primary, recovery, and
 /// confirmation roles.
 #[derive(Debug, uniffi::Object)]
+#[uniffi::export(Hash, Eq)]
 pub struct SecurityShieldBuilder {
     wrapped: RwLock<sargon::SecurityShieldBuilder>,
```
Comment (GhenadieVP) [inline crates/sargon-uniffi/src/profile/mfa/security_structures/security_shield_builder.rs:24]: Do we still need RwLock?

Comment (danvleju-rdx) [reply crates/sargon-uniffi/src/profile/mfa/security_structures/security_shield_builder.rs]: No, removed

File: crates/sargon/src/profile/mfa/security_structures/security_shield_builder.rs [hunk 2]
```diff
@@ -53,6 +53,81 @@ impl Default for SecurityShieldBuilder {
     }
 }
 
+impl PartialEq for SecurityShieldBuilder {
+    fn eq(&self, other: &Self) -> bool {
+        let (matrix, name, authentication_signing_factor) = (
+            self.matrix_builder
+                .read()
+                .expect("Failed to read matrix_builder"),
+            self.name.read().expect("Failed to read name"),
```
Comment (CyonAlexRDX) [inline crates/sargon/src/profile/mfa/security_structures/security_shield_builder.rs:62]: we might wanna use `try_read` here? but we can merge without. Neither Eq nor Clone impl should blocking wait

**Review by CyonAlexRDX — APPROVED** [review]
Maybe not blocking wait in clone and eq.

**Review by sergiupuhalschi-rdx — APPROVED** [review]

**Review by GhenadieVP — APPROVED** [review]

**Pull request merged**